test/e2e/common/util.go

```diff
@@ -184,3 +186,60 @@ func RestartNodes(c clientset.Interface, nodes []v1.Node) error {
 	}
 	return nil
 }
+
+func RecreateNodes(c clientset.Interface, nodes []v1.Node) error {
+	// Build mapping from zone to nodes in that zone.
+	nodeNamesByZone := make(map[string][]string)
+	for i := range nodes {
+		node := &nodes[i]
+		zone := framework.TestContext.CloudConfig.Zone
+		if z, ok := node.Labels[v1.LabelZoneFailureDomain]; ok {
+			zone = z
+		}
+		nodeNamesByZone[zone] = append(nodeNamesByZone[zone], node.Name)
+	}
+
+	// Find the sole managed instance group name
+	var instanceGroup string
+	if strings.Index(framework.TestContext.CloudConfig.NodeInstanceGroup, ",") >= 0 {
+		framework.Failf("Test does not support cluster setup with more than one managed instance group: %s", framework.TestContext.CloudConfig.NodeInstanceGroup)
```

Comment:
I think it would be more consistent to just return an error here rather than failing the test from within a helper function. The caller should handle the error appropriately.

Reply:
Thanks